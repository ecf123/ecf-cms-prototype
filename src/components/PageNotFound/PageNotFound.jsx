import "./PageNotFound.scss";
import magnifyingGlass from "../../assets/images/404-not-found-magnifying-glass.png";
import Home from "../../pages/Home/Home";


const PageNotFound = () => {

  return (
    <div className="error">
    <img className="error__image" src={magnifyingGlass} alt="A magnifying glass"/>
    <h4 className="error__message" >Sorry the page you requested could not be found.</h4>
    <h5 className="error__message">Please go back to the home page</h5>
    <button className="error__button" onClick={<Home />}>Home</button>
    
   </div>
  )
}

export default PageNotFound