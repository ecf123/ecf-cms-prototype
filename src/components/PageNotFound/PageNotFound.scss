@use "../../styles/_variables" as *;
@use "../../styles/_typography" as *;


.pageNotFound{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;

    &__image{
        width: 500px;
        height: auto;
    }

    &__message{
        color: $color-blue-navy;
    }

    &__button{
        margin: 0.5rem;
        width: 6rem;
        height: 2.5rem;
        background-color: #004273;
        border: none;
        font-size: 1.5rem;
        border-radius: 2rem;
        cursor: pointer;
    }

}
