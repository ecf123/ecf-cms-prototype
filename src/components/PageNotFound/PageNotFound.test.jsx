import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import { BrowserRouter } from "react-router-dom";
import PageNotFound from "./PageNotFound";

it("Should render the Page Not Found ", () => {
  const { container } = customRender(<PageNotFound />);

  expect(container).toMatchSnapshot();
});

it("Should render an image", () => {
  render(<PageNotFound />, { wrapper: BrowserRouter });

  const image = screen.getByRole("img");
  expect(image).toBeInTheDocument();
});

it("Should render a button", () => {
  render(<PageNotFound />, { wrapper: BrowserRouter });

  const button = screen.getByRole("button");
  expect(button).toBeInTheDocument();
});
