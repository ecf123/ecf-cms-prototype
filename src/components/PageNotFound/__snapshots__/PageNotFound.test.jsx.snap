// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Page Not Found  1`] = `
<div>
  <div
    class="pageNotFound"
  >
    <img
      alt="A magnifying glass"
      class="pageNotFound__image"
      src="404-not-found-magnifying-glass.png"
    />
    <h4
      class="pageNotFound__message"
    >
      Sorry the page you requested could not be found.
    </h4>
    <h5
      class="pageNotFound__message"
    >
      Please go back to the home page
    </h5>
    <a
      href="/*"
    >
      <button
        class="pageNotFound__button"
      >
        Home
      </button>
    </a>
  </div>
</div>
`;
